### PR TITLE
introduce shared IHCollection tests for IHList, IHSet, and IHQueue.

### PR DESCRIPTION
**Pull request by asimarslan**
API doc improvement
fix API methods

### CODE REVIEW

File: src/Hazelcast.Net.Testing/SingleMemberClientRemoteTestBase.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public async Task ClientOneTimeTearDown()
         /// <summary>
         /// Gets the Hazelcast client.
         /// </summary>
-        public IHazelcastClient Client { get; private set; }
+        public static IHazelcastClient Client { get; private set; }
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net.Testing/SingleMemberClientRemoteTestBase.cs:43]: Not sure I like this. Anything static in test classes and especially in test *base* classes can lead to collisions between tests. Can it remain an instance variable?

File: src/Hazelcast.Net/DistributedObjects/IHList.cs [hunk 2]
```diff
@@ -34,28 +41,29 @@ public interface IHList<T> : IHCollection<T>
...
         /// <summary>
-        /// Replaces the element at the specified position in this list with the
-        /// specified element.
+        /// Replaces the element at the specified position in this list with the specified element.
         /// </summary>
         /// <param name="index">index index of the element to replace</param>
         /// <param name="item">element to be stored at the specified position</param>
-        /// <returns></returns>
+        /// <returns>The element previously at the specified position</returns>
         Task<T> SetAsync(int index, T item);
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net/DistributedObjects/IHList.cs:67]: If we return a value should this be `GetAndSetAsync` just to be consistent?

File: src/Hazelcast.Net.Tests/Remote/ClientBaseCollectionTest.cs [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Hazelcast.DistributedObjects;
+using Hazelcast.Testing;
+using NUnit.Framework;
+
+namespace Hazelcast.Tests.Remote
+{
+    public abstract class ClientBaseCollectionTest : SingleMemberClientRemoteTestBase
```
Comment (zpqrtbnk) [inline src/Hazelcast.Net.Tests/Remote/ClientBaseCollectionTest.cs:25]: This is a base, abstract class → should be something like `ClientCollectionTestBase`.

**Review by zpqrtbnk — COMMENTED** [review]
A few comments, if you don't mind.

File: src/Hazelcast.Net.Testing/SingleMemberClientRemoteTestBase.cs (hunk 1 above)
Comment (asimarslan) [reply src/Hazelcast.Net.Testing/SingleMemberClientRemoteTestBase.cs:43]: ok

File: src/Hazelcast.Net.Tests/Remote/ClientBaseCollectionTest.cs (hunk 3 above)
Comment (asimarslan) [reply src/Hazelcast.Net.Tests/Remote/ClientBaseCollectionTest.cs:25]: ok

File: src/Hazelcast.Net/DistributedObjects/IHList.cs (hunk 2 above)
Comment (asimarslan) [reply src/Hazelcast.Net/DistributedObjects/IHList.cs:67]: ok changing the RemoveAt too

**Review by zpqrtbnk — APPROVED** [review]
all good

**Pull request merged**